packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -245,7 +264,11 @@ export class HardhatNode extends EventEmitter {
       genesisAccounts,
       tracingConfig,
       forkNetworkId,
-      nextBlockBaseFeePerGas
+      nextBlockBaseFeePerGas,
+      hardforkActivationBlocks !== undefined &&
+      hardforkActivationBlocks[chainId] !== undefined
```

Comment:
Use `forkNetworkId`, not `chainId`, because it's only used for the remote one.

Reply:
Can possibly use `hardforkActivationBlocks?[chainId] ?? undefined`.  Or will `?[` work? Isn't that just `?.`? but isn't `[x]` the same as `.x`?

Reply:
https://github.com/nomiclabs/hardhat/issues/2015